### PR TITLE
Add travis-ci build

### PR DESCRIPTION
**Pull request by confusdcodr**

### CODE REVIEW

File: build/install_linux.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+curl -o bootstrap-salt.sh -L https://bootstrap.saltstack.com
+sudo sh bootstrap-salt.sh
+sudo apt install git-all
```
Comment (lorengordon) [inline build/install_linux.sh]: Can remove this, `git` is already installed in the build env

File: build/install_linux.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+git clone https://github.com/plus3it/amazon-inspector-formula
```
Comment (lorengordon) [inline build/install_linux.sh]: Can remove this, travis handles it automatically, always. The current working dir is already the project root.

File: build/install_linux.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+sudo mkdir -p /srv/salt
+sudo cp -r amazon-inspector-formula/amazon-inspector /srv/salt/
```
Comment (lorengordon) [inline build/install_linux.sh]: Might need to revise this path, probably:

```suggestion
sudo cp -r amazon-inspector /srv/salt/
```

File: build/install_windows.ps1 [hunk 4]
```diff
@@ -0,0 +1,8 @@
+Set-StrictMode -Version Latest
+$ErrorActionPreference = "Stop"
+$PSDefaultParameterValues['*:ErrorAction']='Stop'
+
+Set-Location "$Env:TRAVIS_BUILD_DIR"
+git clone "https://github.com/plus3it/amazon-inspector-formula"
```
Comment (lorengordon) [inline build/install_windows.ps1]: Same deal, no need to re-clone the project, travis has done this already.

File: build/install_windows.ps1 [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+$saltDir = "C:\salt\srv\salt\"
+Copy-Item "$Env:TRAVIS_BUILD_DIR\amazon-inspector-formula\" -Destination $saltDir -Recurse
```
Comment (lorengordon) [inline build/install_windows.ps1]: Not sure on `Copy-Item` semantics, but same idea, may need to adjust the copy path

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  include:
+    - os: windows
+      language: sh
+      python: "3.7"
+      before_install:
+        - choco install saltminion
+        - export PATH="/c/salt:$PATH"
+
+install:
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then bash ./build/install_linux.sh; fi
```
Comment (lorengordon) [inline .travis.yml]: need to use `bash -e` to avoid false successes

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  - if [[ "$TRAVIS_OS_NAME" == "windows" ]]; then powershell -ExecutionPolicy RemoteSigned ./build/install_windows.ps1; fi
+
+script:
+  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then bash ./build/script_linux.sh; fi
```
Comment (lorengordon) [inline .travis.yml]: `bash -e`

File: build/script_windows.ps1 [hunk 8]
```diff
@@ -0,0 +1,6 @@
+Set-StrictMode -Version Latest
+$ErrorActionPreference = "Stop"
+$PSDefaultParameterValues['*:ErrorAction']='Stop'
+
+Invoke-Expression -Command "cmd.exe /c salt-call --local state.show_sls amazon-inspector"
```
Comment (lorengordon) [inline build/script_windows.ps1]: Please confirm that this actually errexits by testing with a known-failing command...

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,21 @@
+---
+language: python
+python:
+  - "3.5"
```
Comment (lorengordon) [inline .travis.yml]: Any particular requirement for the extra python build?

Comment (confusdcodr) [reply .travis.yml]: Wasn't able to get the Windows build to be recognized without it

File: build/script_windows.ps1 (hunk 8 above)
Comment (confusdcodr) [reply build/script_windows.ps1]: The script will completely stop running and throw the error. It will not continue on to the other commands.

**Comment by confusdcodr** [comment]
Travis jobs succeed - https://travis-ci.com/plus3it/amazon-inspector-formula/jobs/172843401

File: build/script_windows.ps1 (hunk 8 above)
Comment (lorengordon) [reply build/script_windows.ps1]: Ok thanks. One more favor... Can you introduce a syntax error into the formula itself and confirm that travis fails?

File: .travis.yml (hunk 9 above)
Comment (lorengordon) [reply .travis.yml]: Still? Oh, I thought that was resolved.

Comment (lorengordon) [reply .travis.yml]: This doc helps... https://docs.travis-ci.com/user/customizing-the-build/#explicitly-including-jobs

I'd expect you could eliminate the top-level `python` key, and just use the matrix:

```
language: python
matrix:
  include:
    - os: linux
      python: 3.6
    - os: windows
      language: sh
      python: 3.7
      before_install:
        - choco install saltminion
        - export PATH="/c/salt:$PATH"
```

This page is also very helpful: https://docs.travis-ci.com/user/multi-os/

Comment (confusdcodr) [reply .travis.yml]: That worked fine, thanks! I guess having the python version on its own is what caused the issue previously.

File: build/script_windows.ps1 (hunk 8 above)
Comment (confusdcodr) [reply build/script_windows.ps1]: Looks like the intentional formula error is being masked - https://travis-ci.com/confusdcodr/amazon-inspector-formula/builds/98816149

Comment (lorengordon) [reply build/script_windows.ps1]: This option is necessary for salt to pass through failures in state execution to the return code of salt itself: `--retcode-passthrough`

Comment (confusdcodr) [reply build/script_windows.ps1]: Successfully forced travis to fail with intentional error - https://travis-ci.com/confusdcodr/amazon-inspector-formula/builds/98823446

Comment (confusdcodr) [reply build/script_windows.ps1]: And now we are successfully building again - https://travis-ci.com/confusdcodr/amazon-inspector-formula/builds/98824534

File: .travis.yml [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        - sudo salt-call --local --retcode-passthrough state.show_sls amazon-inspector
+        - sudo salt-call --local --retcode-passthrough state.sls amazon-inspector mock=True 
+    - os: windows
+      language: sh
+      python: 3.7
+      before_install:
+        - choco install saltminion
+        - export PATH="/c/salt:$PATH"
+      install:
+        - powershell -ExecutionPolicy RemoteSigned ./build/install_windows.ps1
```
Comment (lorengordon) [inline .travis.yml]: We ought to be able to just use `cp` via git bash to copy the files...

```
cp -r "amazon-inspector" /c/salt/srv/salt
```

Comment (confusdcodr) [reply .travis.yml]: Done.

**Review by lorengordon — APPROVED** [review]

**Pull request merged**